astropy/cosmology/tests/test_funcs.py

```diff
@@ -233,8 +234,9 @@ def test_z_at_value_roundtrip(cosmo):
     if str(cosmo.name).startswith('WMAP'):
         skip += ('nu_relative_density', )
 
-    import inspect
-    methods = inspect.getmembers(cosmo, predicate=inspect.ismethod)
+    # get methods, ignoring warning from ".read"
```

Comment:
Ah, I guess `AttributeError` might make it impossible to do this inspection...

Reply:
This was the initial reason for raising a warning over an error. I also didn't like it philosophically, that an uninspectable "attribute" could exist on a class.

Reply:
OK, since you plan to change it anyway later, let's just stick with what you have for now!